spyder/plugins/editor/widgets/main_widget.py

```diff
@@ -0,0 +1,3586 @@
+        editorstack.sig_next_cursor.connect(self.go_to_next_cursor_position)
+        editorstack.sig_prev_warning.connect(self.go_to_previous_warning)
+        editorstack.sig_next_warning.connect(self.go_to_next_warning)
+        editorstack.sig_save_bookmark.connect(self.save_bookmark)
+        editorstack.sig_load_bookmark.connect(self.load_bookmark)
+        editorstack.sig_save_bookmarks.connect(self.save_bookmarks)
+        editorstack.sig_help_requested.connect(self.sig_help_requested)
+        editorstack.sig_codeeditor_created.connect(self.sig_codeeditor_created)
+        editorstack.sig_codeeditor_changed.connect(self.sig_codeeditor_changed)
+        editorstack.sig_codeeditor_deleted.connect(self.sig_codeeditor_deleted)
+
+        # Register editorstack's autosave component with plugin's autosave
+        # component
+        self.autosave.register_autosave_for_stack(editorstack.autosave)
+
+    def unregister_editorstack(self, editorstack):
+        """Removing editorstack only if it's not the last remaining"""
+        logger.debug("Unregistering EditorStack")
+        self.remove_last_focused_editorstack(editorstack)
+        if len(self.editorstacks) > 1:
+            index = self.editorstacks.index(editorstack)
+            self.editorstacks.pop(index)
+            self.find_widget.set_editor(self.get_current_editor())
+            return True
+        else:
+            # editorstack was not removed!
+            return False
+
+    def clone_editorstack(self, editorstack):
+        editorstack.clone_from(self.editorstacks[0])
+
+    @Slot(str, str)
+    def close_file_in_all_editorstacks(self, editorstack_id_str, filename):
+        if filename in self.id_per_file:
+            self.deregister_file_run_metadata(filename)
+        else:
+            _, filename_ext = osp.splitext(filename)
+            filename_ext = filename_ext[1:]
+            self.pending_run_files -= {(filename, filename_ext)}
+
+        try:
+            for editorstack in self.editorstacks:
+                if str(id(editorstack)) != editorstack_id_str:
+                    editorstack.blockSignals(True)
+                    index = editorstack.get_index_from_filename(filename)
+                    editorstack.close_file(index, force=True)
+                    editorstack.blockSignals(False)
+        except RuntimeError:
+            pass
+
+    @Slot(str, str, str)
+    def file_saved_in_editorstack(self, editorstack_id_str,
+                                  original_filename, filename):
+        """A file was saved in editorstack, this notifies others"""
+        for editorstack in self.editorstacks:
+            if str(id(editorstack)) != editorstack_id_str:
+                editorstack.file_saved_in_other_editorstack(original_filename,
+                                                            filename)
+
+    # ---- Handling editor windows
+    # TODO: args added as a workaround, should be removed. Maybe
+    # this should be moved to the plugin?
```

Comment:
Change to use actions registry done at https://github.com/spyder-ide/spyder/pull/19784 (fix future conflict there)